### ISSUE TITLE
Install problem

### ISSUE THREAD

**Issue by mathxiaobin** [Wuziyi616/multi_part_assembly#12]
Hello, when i install this project following the install.md(https://github.com/Wuziyi616/multi_part_assembly/blob/master/docs/install.md). i allways face a problem in the step "Custom Ops" as blew:

Obtaining file:///home/xb/ProjectToy/multi_part_assembly-master/multi_part_assembly/utils/chamfer
  Preparing metadata (setup.py) ... done
DEPRECATION: pytorch-lightning 1.6.2 has a non-standard dependency specifier torch>=1.8.*. pip 23.3 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of pytorch-lightning or contact the author to suggest that they release a version with a conforming dependency specifiers. Discussion can be found at https://github.com/pypa/pip/issues/12063
Installing collected packages: chamfer-ext
  Running setup.py develop for chamfer-ext
    error: subprocess-exited-with-error

    × python setup.py develop did not run successfully.
    │ exit code: 1
    ╰─> [39 lines of output]
        running develop
        running egg_info
        creating chamfer_ext.egg-info
        writing chamfer_ext.egg-info/PKG-INFO
        writing dependency_links to chamfer_ext.egg-info/dependency_links.txt
        writing top-level names to chamfer_ext.egg-info/top_level.txt
        writing manifest file 'chamfer_ext.egg-info/SOURCES.txt'
        reading manifest file 'chamfer_ext.egg-info/SOURCES.txt'
        writing manifest file 'chamfer_ext.egg-info/SOURCES.txt'
        running build_ext
        /home/xb/anaconda3/envs/assembly/lib/python3.8/site-packages/setuptools/command/develop.py:40: EasyInstallDeprecationWarning: easy_install command is deprecated.
        !!

                ********************************************************************************
                Please avoid running ``setup.py`` and ``easy_install``.
                Instead, use pypa/build, pypa/installer or other
                standards-based tools.

                See https://github.com/pypa/setuptools/issues/917 for details.
                ********************************************************************************

        !!
          easy_install.initialize_options(self)
        /home/xb/anaconda3/envs/assembly/lib/python3.8/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
        !!

                ********************************************************************************
                Please avoid running ``setup.py`` directly.
                Instead, use pypa/build, pypa/installer or other
                standards-based tools.

                See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
                ********************************************************************************

        !!
          self.initialize_options()
        /home/xb/anaconda3/envs/assembly/lib/python3.8/site-packages/torch/utils/cpp_extension.py:381: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
          warnings.warn(msg.format('we could not find ninja.'))
        error: [Errno 2] No such file or directory: ':/usr/local/cuda-11.4/bin/nvcc'
        [end of output]

    note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× python setup.py develop did not run successfully.
│ exit code: 1
╰─> [39 lines of output]
    running develop
    running egg_info
    creating chamfer_ext.egg-info
    writing chamfer_ext.egg-info/PKG-INFO
    writing dependency_links to chamfer_ext.egg-info/dependency_links.txt
    writing top-level names to chamfer_ext.egg-info/top_level.txt
    writing manifest file 'chamfer_ext.egg-info/SOURCES.txt'
    reading manifest file 'chamfer_ext.egg-info/SOURCES.txt'
    writing manifest file 'chamfer_ext.egg-info/SOURCES.txt'
    running build_ext
    /home/xb/anaconda3/envs/assembly/lib/python3.8/site-packages/setuptools/command/develop.py:40: EasyInstallDeprecationWarning: easy_install command is deprecated.
    !!

            ********************************************************************************
            Please avoid running ``setup.py`` and ``easy_install``.
            Instead, use pypa/build, pypa/installer or other
            standards-based tools.

            See https://github.com/pypa/setuptools/issues/917 for details.
            ********************************************************************************

    !!
      easy_install.initialize_options(self)
    /home/xb/anaconda3/envs/assembly/lib/python3.8/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
    !!

            ********************************************************************************
            Please avoid running ``setup.py`` directly.
            Instead, use pypa/build, pypa/installer or other
            standards-based tools.

            See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
            ********************************************************************************

    !!
      self.initialize_options()
    /home/xb/anaconda3/envs/assembly/lib/python3.8/site-packages/torch/utils/cpp_extension.py:381: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
      warnings.warn(msg.format('we could not find ninja.'))
    error: [Errno 2] No such file or directory: ':/usr/local/cuda-11.4/bin/nvcc'
    [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.

could you please give me some guidance to solve it？

**Comment by Wuziyi616** [comment]
Can you show me your output of `nvcc --version` or `which nvcc`? This custom ops require CUDA to compile it, which needs you to install cudatoolkit and the `nvcc` bin file.

**Comment by mathxiaobin** [comment]
> Can you show me your output of `nvcc --version` or `which nvcc`? This custom ops require CUDA to compile it, which needs you to install cudatoolkit and the `nvcc` bin file.

My "nvcc --version" and "nvidia-smi" outputs are :

(assembly) xb@ThinkCentre:~$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Wed_Jun__2_19:15:15_PDT_2021
Cuda compilation tools, release 11.4, V11.4.48
Build cuda_11.4.r11.4/compiler.30033411_0
(assembly) xb@ThinkCentre:~$ nvidia-smi
Sat Aug 19 09:44:15 2023
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.82.01    Driver Version: 470.82.01    CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:01:00.0 N/A |                  N/A |
| 25%   39C    P8    N/A /  N/A |     93MiB /  1985MiB |     N/A      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
(assembly) xb@ThinkCentre:~$ which nvcc
/usr/local/cuda-11.4/bin/nvcc


thanks!

**Comment by mathxiaobin** [comment]
> Can you show me your output of `nvcc --version` or `which nvcc`? This custom ops require CUDA to compile it, which needs you to install cudatoolkit and the `nvcc` bin file.

My "nvcc --version" and "nvidia-smi" outputs are :

(assembly) xb@ThinkCentre:~$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Wed_Jun__2_19:15:15_PDT_2021
Cuda compilation tools, release 11.4, V11.4.48
Build cuda_11.4.r11.4/compiler.30033411_0
(assembly) xb@ThinkCentre:~$ which nvcc
/usr/local/cuda-11.4/bin/nvcc
(assembly) xb@ThinkCentre:~$ nvidia-smi
Sat Aug 19 09:44:15 2023
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.82.01    Driver Version: 470.82.01    CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:01:00.0 N/A |                  N/A |
| 25%   39C    P8    N/A /  N/A |     93MiB /  1985MiB |     N/A      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

**Comment by Wuziyi616** [comment]
Hummm that's weird because your nvcc is indeed at that location. I've never met such a problem before so I'm not sure if I can fix it. Which pytorch version are you using?

**Comment by mathxiaobin** [comment]
> Hummm that's weird because your nvcc is indeed at that location. I've never met such a problem before so I'm not sure if I can fix it. Which pytorch version are you using?

my pytorch version is "pytorch  1.10.2  py3.8_cuda11.3_cudnn8.2.0_0   pytorch"

Here are my conda environment details:

(assembly) xb@ThinkCentre:~$ conda list
# packages in environment at /home/xb/anaconda3/envs/assembly:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             5.1                       1_gnu
absl-py                   1.4.0                    pypi_0    pypi
aiohttp                   3.8.5                    pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
async-timeout             4.0.3                    pypi_0    pypi
attrs                     23.1.0                   pypi_0    pypi
blas                      1.0                         mkl
bzip2                     1.0.8                h7b6447c_0
ca-certificates           2023.7.22            hbcca054_0    conda-forge
cachetools                5.3.1                    pypi_0    pypi
certifi                   2023.7.22                pypi_0    pypi
charset-normalizer        3.2.0                    pypi_0    pypi
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
cudatoolkit               11.3.1               h2bc3f7f_2
ffmpeg                    4.3                  hf484d3e_0    pytorch
freetype                  2.12.1               h4a9f257_0
frozenlist                1.4.0                    pypi_0    pypi
fsspec                    2023.6.0                 pypi_0    pypi
fvcore                    0.1.5.post20210915            py38    fvcore
giflib                    5.2.1                h5eee18b_3
gmp                       6.2.1                h295c915_3
gnutls                    3.6.15               he1e5248_0
google-auth               2.22.0                   pypi_0    pypi
google-auth-oauthlib      1.0.0                    pypi_0    pypi
grpcio                    1.57.0                   pypi_0    pypi
idna                      3.4                      pypi_0    pypi
importlib-metadata        6.8.0                    pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46305
iopath                    0.1.9                      py38    iopath
jpeg                      9e                   h5eee18b_1
lame                      3.100                h7b6447c_0
lcms2                     2.12                 h3be6417_0
ld_impl_linux-64          2.38                 h1181459_1
lerc                      3.0                  h295c915_0
libdeflate                1.17                 h5eee18b_0
libffi                    3.4.4                h6a678d5_0
libgcc-ng                 11.2.0               h1234567_1
libgomp                   11.2.0               h1234567_1
libiconv                  1.16                 h7f8727e_2
libidn2                   2.3.4                h5eee18b_0
libpng                    1.6.39               h5eee18b_0
libstdcxx-ng              11.2.0               h1234567_1
libtasn1                  4.19.0               h5eee18b_0
libtiff                   4.5.0                h6a678d5_2
libunistring              0.9.10               h27cfd23_0
libuv                     1.44.2               h5eee18b_0
libwebp                   1.2.4                h11a3e52_1
libwebp-base              1.2.4                h5eee18b_1
lightning-utilities       0.9.0                    pypi_0    pypi
lz4-c                     1.9.4                h6a678d5_0
markdown                  3.4.4                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46343
mkl-service               2.4.0            py38h5eee18b_1
mkl_fft                   1.3.6            py38h417a72b_1
mkl_random                1.2.2            py38h417a72b_1
multidict                 6.0.4                    pypi_0    pypi
ncurses                   6.4                  h6a678d5_0
nettle                    3.7.3                hbbd107a_1
numpy                     1.24.3           py38hf6e8229_1
numpy-base                1.24.3           py38h060ed82_1
oauthlib                  3.2.2                    pypi_0    pypi
openh264                  2.1.1                h4ff587b_0
openssl                   3.0.10               h7f8727e_0
pathtools                 0.1.2                    pypi_0    pypi
pillow                    9.4.0            py38h6a678d5_0
pip                       23.2.1           py38h06a4308_0
portalocker               1.4.0                      py_0    conda-forge
protobuf                  4.24.0                   pypi_0    pypi
pyasn1                    0.5.0                    pypi_0    pypi
pyasn1-modules            0.3.0                    pypi_0    pypi
pydeprecate               0.3.2                    pypi_0    pypi
python                    3.8.17               h955ad1f_0
pytorch                   1.10.2          py3.8_cuda11.3_cudnn8.2.0_0    pytorch
pytorch-lightning         1.6.2                    pypi_0    pypi
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2023.3                   pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
readline                  8.2                  h5eee18b_0
requests                  2.31.0                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
scipy                     1.10.1                   pypi_0    pypi
sentry-sdk                1.29.2                   pypi_0    pypi
setproctitle              1.3.2                    pypi_0    pypi
setuptools                68.0.0           py38h06a4308_0
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.0                    pypi_0    pypi
sqlite                    3.41.2               h5eee18b_0
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tbb                       2021.8.0             hdb19cb5_0
tensorboard               2.14.0                   pypi_0    pypi
tensorboard-data-server   0.7.1                    pypi_0    pypi
termcolor                 2.3.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h1ccaba5_0
torchaudio                0.10.2               py38_cu113    pytorch
torchmetrics              1.0.3                    pypi_0    pypi
torchvision               0.11.3               py38_cu113    pytorch
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
trimesh                   3.23.3                   pypi_0    pypi
typing_extensions         4.7.1            py38h06a4308_0
urllib3                   1.26.16                  pypi_0    pypi
werkzeug                  2.3.7                    pypi_0    pypi
wheel                     0.38.4           py38h06a4308_0
xz                        5.4.2                h5eee18b_0
yacs                      0.1.8              pyhd8ed1ab_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
yarl                      1.9.2                    pypi_0    pypi
zlib                      1.2.13               h5eee18b_0
zstd                      1.5.5                hc292b87_0

**Comment by Wuziyi616** [comment]
Please try an earlier version of PyTorch, see [here](https://github.com/Wuziyi616/multi_part_assembly/blob/master/docs/install.md#custom-ops), PyTorch 1.10.2 has some weird issues.

**Comment by mathxiaobin** [comment]
> Please try an earlier version of PyTorch, see [here](https://github.com/Wuziyi616/multi_part_assembly/blob/master/docs/install.md#custom-ops), PyTorch 1.10.2 has some weird issues.

I reinstall the CUDA and pytorch, but it dose not works.

Finally, I discovered a low-level error. In the initial error, there was a sentence of "No such file or directory: ':/usr/local/cuda-11.4/bin/nvcc'", and I believe that the ":"  before “/usr” was the source of the error. Therefore, I changed the environment variable in the "bashrc" file from "export CUDA_HOME=$CUDA_HOME:/usr/local/cuda-11.3" to "export CUDA_HOME=/usr/local/cuda-11.3". Afterwards, the installation of Custom Ops is correct.

Thank you for your reply and help!
